agent/config/config_accessor.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"errors"
+)
+
+// agentConfigAccessor struct implements AgentConfigAccessor interface defined in ecs-agent module.
+type agentConfigAccessor struct {
+	cfg *Config
+}
+
+// NewAgentConfigAccessor creates a new agentConfigAccessor.
+func NewAgentConfigAccessor(cfg *Config) (*agentConfigAccessor, error) {
+	if cfg == nil {
+		return nil, errors.New("unable to create new agent config accessor due to passed in config value being nil")
+	}
+
+	return &agentConfigAccessor{cfg: cfg}, nil
+}
+
+func (aca *agentConfigAccessor) APIEndpoint() string {
+	return aca.cfg.APIEndpoint
+}
+
+func (aca *agentConfigAccessor) AWSRegion() string {
+	return aca.cfg.AWSRegion
+}
+
+func (aca *agentConfigAccessor) Cluster() string {
+	return aca.cfg.Cluster
+}
+
+func (aca *agentConfigAccessor) DefaultClusterName() string {
+	return DefaultClusterName
+}
+
+func (aca *agentConfigAccessor) External() bool {
+	return aca.cfg.External.Enabled()
+}
+
+func (aca *agentConfigAccessor) InstanceAttributes() map[string]string {
+	return aca.cfg.InstanceAttributes
+}
+
+func (aca *agentConfigAccessor) NoInstanceIdentityDocument() bool {
+	return aca.cfg.NoIID
+}
+
+func (aca *agentConfigAccessor) OSFamily() string {
+	return GetOSFamily()
+}
+
+func (aca *agentConfigAccessor) OSType() string {
+	return OSType
```

Comment:
things that are not in the config, do we need these getter methods? or should them be in the config?

Reply:
We need these getters as these things are defined in agent module in the `agent/config/` directory. The `AgentConfigAccessor` interface will be used in ecs-agent module and that interface requires access to such agent-level information. Thus, we need this implementation to have the corresponding methods to access thest things. 

`OSType` and `GetOSFamily()` seem to be determined based off go build flags such as the ones defined [here](https://github.com/aws/amazon-ecs-agent/blob/dev/agent/config/parse_unsupported.go#L1-L2) (as an example). Since `OSType` and `GetOSFamily()` are determined initially and shouldn't be modified after, I don't think they should be in the `Config` struct itself but rather their separate own things in the agent/config/ directory, as we would want to keep the determined values and not have those values be configurable.